openmls/src/treesync/node/leaf_node.rs

```diff
@@ -114,6 +205,27 @@ impl LeafNode {
         Ok((leaf_node, encryption_key_pair))
     }
 
+    /// Creates a new placeholder [`LeafNode`] that is used to build external
+    /// commits.
+    ///
+    /// Note: This is not a valid leaf node and it must be rekeyed and signed
+    /// before it can be used.
+    pub(crate) fn new_placeholder() -> Self {
+        let payload = LeafNodePayload {
+            encryption_key: EncryptionKey::from(Vec::new()),
+            signature_key: Vec::new().into(),
+            credential: Credential::new(CredentialType::Basic, Vec::new()),
+            capabilities: Capabilities::default(),
+            leaf_node_source: LeafNodeSource::Update,
+            extensions: Extensions::default(),
+        };
+
+        Self {
+            payload,
+            signature: Vec::new().into(),
+        }
+    }
+
```

Comment:
I suppose we can't easily prevent using this in the wrong places using the type system, right?

Reply:
The only way I saw was to refactor large parts of TreeSync to accommodate that short-lived temporary state. I'm a big fan of type safety, but that would not be reasonable at this point.